gapis/service/service.proto

```diff
@@ -671,6 +672,76 @@ message MemoryRange {
   uint64 size = 2;
 }
 
+message Metrics {
+  oneof metrics {
+    MemoryLayout memory_layout = 1;
+  }
+}
+
+// The description of the memory layout of the API state
+message MemoryLayout {
+  // The API ID used for this call.
+  path.API api = 1;
+  // The index to get the constants names for MemoryAllocation `flags`.
+  // A value of -1 indicates no flag names should be fetched.
+  int32 allocation_flags_index = 2;
+  // The memory allocations that are currently in use.
+  repeated MemoryAllocation allocations = 3;
+}
+
+// A single memory allocation
+message MemoryAllocation {
+  // The device this allocation was made on
+  uint64 device = 1;
+  // The memory type this allocation is from
+  uint32 memory_type = 2;
```

Comment:
Looks like this is more like `memory_type_index`? But `index` also sounds too Vulkan specific, I'm not sure if in GL world there is a thing like `type index`.

Reply:
Yes, it is memory type index in Vulkan, I didn't want to make it overspecific to Vulkan though.

Reply:
GL works a bit differently, since all memory is implicit. The allocations will be 1:1 with the resources.
I would expect something like "memory_type" to be "GL_IMAGE or GL_BUFFER" most likely.